api/app.py

```diff
@@ -0,0 +1,96 @@
+import connexion
+import os
+import json
+import logging
+from jupyterhub_singleuser_profiles.profiles import SingleuserProfiles
+
+from functools import wraps
+from urllib.parse import quote
+
+from flask import Flask
+from flask import redirect
+from flask import request
+from flask import Response
+
+from jupyterhub.services.auth import HubAuth
+
+_PROFILES = SingleuserProfiles(verify_ssl=False)
+_PROFILES.load_profiles()
+_PATH = "/opt/app-root/lib64/python3.6/site-packages"
+
+_LOGGER = logging.getLogger(__name__)
+
+prefix = os.environ.get('JUPYTERHUB_SERVICE_PREFIX', '/')
+
+auth = HubAuth(api_token=os.environ['JUPYTERHUB_API_TOKEN'], cache_max_age=60)
+
+app = Flask(__name__)
+
+def authenticated(f):
+    """Decorator for authenticating with the Hub"""
+
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        cookie = request.cookies.get(auth.cookie_name)
+        token = request.headers.get(auth.auth_header_name)
+        for_user = connexion.request.headers['For-User']
```

Comment:
This should be 
```
for_user = connexion.request.headers.get('For-User')
```

`.get` will not fail if the header does not exist and will set `None` as default